### DISCUSSION TITLE
Lession 10: Where to find the interface for WETH - https://youtu.be/M576WGiDBdQ?t=31970

### DISCUSSION THREAD

**Discussion by rsamban** [smartcontractkit/full-blockchain-solidity-course-py#640] [Q&A] [answered]
At https://youtu.be/M576WGiDBdQ?t=31970 , Patrck says that he already copied the interface [here](https://github.com/PatrickAlphaC/aave_brownie_py/blob/main/interfaces/WethInterface.sol). 

https://kovan.etherscan.io/token/0xd0a1e359811322d97991e03f863a0c30c2cf029c#readContract does not have the contract code.

Also, the interface code in IWETH.sol in Wethgateway contract (https://kovan.etherscan.io/address/0x6382ab06cdeafd25162699e053fafad1aca68d23#code) does not have all the functions defined in [here](https://github.com/PatrickAlphaC/aave_brownie_py/blob/main/interfaces/WethInterface.sol)

Trying to find where(git repo) he copied the interface code from...

**Comment by cromewar** [comment, marked as answer]
Hello @rsamban I'm pretty much sure he created that interface on his own based on the standard for ERC20 tokens, as you can see on link_token interface from chainlink: https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/interfaces/LinkTokenInterface.sol so is pretty much based on it.

**Comment by rsamban** [comment]
Thanks @cromewar , but there should be implementation for the functions defined in the interface in some contract, right? How do I find the implementation?

**Reply by cromewar** [reply]
That's the key of the interfaces they don't actually need an implementation, most of the interfaces are meant to be overrided, you can learn more about interfaces in solidity on this link: https://www.tutorialspoint.com/solidity/solidity_interfaces.htm

**Reply by rsamban** [reply]
I guess I get what interfaces are, but when you invoke methods of interfaces, some contract has to have implemented them, (so that at run time it does something useful). In java world, we use interfaces all the time, at runtime we can clearly see that somebody have implemented them with real code.

**Reply by cromewar** [reply]
well it depends, some contracts has their implementations for example ERC20 on Openzeppelin contracts has it interface IERC20, so you are right, on this case the AAVE lesson does not need as it work for a demonstrative  purpose.
